src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -139,7 +140,8 @@ static ProjectCodeRefactoringProvider.ExtensionInfo GetExtensionInfo(ExportCodeR
                             RefactoringToMetadataMap.TryGetValue(provider, out var providerMetadata);
 
                             using (addOperationScope(providerName))
-                            using (RoslynEventSource.LogInformationalBlock(FunctionId.Refactoring_CodeRefactoringService_GetRefactoringsAsync, providerName, cancellationToken))
+                            using (Logger.LogBlock(FunctionId.Refactoring_CodeRefactoringService_GetRefactoringFromProvider, KeyValueLogMessage.Create(LogType.Trace, m => CreateLogProperties(m, provider)), cancellationToken))
+                            using (RoslynEventSource.LogInformationalBlock(FunctionId.Refactoring_CodeRefactoringService_GetRefactoringFromProvider, providerName, cancellationToken))
```

Comment:
Because `GetRefactoringFromProviderAsync` is immedately returning we are just measuring how long it is taking us to setup the task, not how long it runs for. I would move this block inside the `GetRefactoringFromProviderAsync` method so get a real measurement.

Reply:
Done